### ISSUE TITLE
Incorrect call error callback on facebookConnectPlugin.api when user does not give permission

### ISSUE THREAD

**Issue by robertgacki** [jeduan/cordova-plugin-facebook4#145]
Hi,

I think that I found a bug when I call a facebookConnectPlugin.api method . If user removed permission (e.g. to email) and facebookConnectPlugin.api method need this permission then plugin shows facebook prompt with question about permission (there is a good solution) but call error callback with error: 

> Error in facebookConnectPlugin.api me/?fields=id,email This request needs declined permission: email

In my opinion error callback should call when user click cancel on facebook prompt and call success callback when user submit add permission to application.

**Comment by raphaguasta** [comment]
@robertgacki I noticed that too, do you found any solution?

**Comment by robertgacki** [comment]
No. I had to work around this problem. On onResume event I send request to API again.

**Comment by simokhalil** [comment]
Any news about this guys ? same thing here

**Comment by simokhalil** [comment]
I think I got a workaround ! Just add a return in the error test  about declined permission, like so :

`In ConnectPlugin.java:631`

`if (declinedPermission != null) {`
`    graphContext.error("This request needs declined permission: " + declinedPermission);`
`    return;`
`}`

Now, if the user decline a required permission, the handle get stopped and the JS error callback get called without showing the facebook prompt. 
This way, just as described in [this official Facebook doc](https://developers.facebook.com/docs/facebook-login/handling-declined-permissions) (option 2), you can handle the error manually by, for example, showing a dialog explaining to user why you need his data and permitting him to retry, in which case you can recall the .login function to prompt again the user for permissions and finally call .api again to get profile info.

Note : modifying the plugin native files this way is by far not a best practice, as your mod will be overwritten when you will update your plugin, and I don't know if this could be accepted as a PR...

**Comment by fredgalvao** [comment]
I'd like that to be included in the plugin. @simokhalil  If you could create a PR for it, that'd be awesome.

**Comment by simokhalil** [comment]
@fredgalvao Here you are #375

**Comment by robertarnesson** [comment]
I suspect this is not only an android issue?

**Comment by simokhalil** [comment]
@robertarnesson 
I personally only tested on android as my app is only built for android at the moment.
Have you encountred this problem on ios ? I'll take a look at this for ios when I'll have some time next days...

**Comment by robertarnesson** [comment]
I'm guessing its all platforms. I've seen it with the javascript sdk. It would be good to get some feedback from other users and make sure that if we make changes it is the same behaviour on all platforms

**Comment by noahcooper** [comment]
This plugin is deprecated. Check out `cordova-plugin-facebook-connect` at https://www.npmjs.com/package/cordova-plugin-facebook-connect.
